src/module/types/template/Matrix.ts

```diff
@@ -52,6 +53,7 @@ export const MatrixData = () => ({
     link_locked: new BooleanField(),
     hot_sim: new BooleanField(),
     running_silent: new BooleanField(),
+    public_grid_modifier: new NumberField({ required: true, nullable: false, integer: true, initial: MatrixRules.publicGridModifier() }),
```

Comment:
this `public_grid_modifier` is for each item/actor separately. shouldn't it be a global variable? Similar to environment variable

Reply:
I also rather do it through `system.modifiers`, and it also avoids a potential problem with circular import.

Reply:
actually, you would need to refactor the code.
Or assume that the modifier is like `-2 + system.modifier.public_grid_something`...

I haven't coded that expecting a value to have default value different than `0`, sorry... 😓

Reply:
nvm, I realize now the default value will just be '-2' and then apply the modifier, if any

Reply:
Would you guys prefer to 
- 'modify'  the modifier. -2 (const) + 2 (effect change on actor) => 0
- 'override' the modifier 0 (effect change on actor)
?

Personally I'd prefer the last one. I'm not sure if there is fixed const modifier already that is just 'modified' by a `system.modifier` though; in that case, I prefer whatever is already done before.
The 'modify' approach would also need a refactor `Common.ts`#33 => `CreateModifiers`, as Numbers said, as it only accepts a list of modifier names with a default to zero.

However that refactor could simply be
`SchemaField({...CreateModifiers('a', 'b'), ...{c: new NumberField({initial: 1})})`

Reply:
tbh, I don't like either options... it feels there is no right place for it now...

Reply:
Can you elaborate what you dislike?

What would your prefered way as a player be?

Reply:
I think matching what we have now would be option 1: "modify the modifier". I know that it isn't pretty, but I think it best matches how all the other modifiers work in that listing. Nothing else acts as a "replacement" of the value, it modified the existing value. Looking at initiative values.

Reply:
> Can you elaborate what you dislike?
> 
> What would your prefered way as a player be?

It is what @smilligan93 just said. `Modifiers` should not replace the value in that part of the sheet, since no other modifier follows this behavior. And "Modify the modifier" is a bit weird, but I rather do this (modify the modifier) than having a different behavior.

Reply:
Then we're all in agreement :)